### PR TITLE
Only require typing-extensions if Python < 3.8

### PR DESCRIPTION
**Pull request by hukkinj1**
This PR makes Black installation one dependency lighter in Python 3.8+

### CODE REVIEW

**Review by cooperlees — APPROVED** [review]
Seems sane to me. Always nice to remove deps. Thanks!

**Comment by cooperlees** [comment]
My only question is do we need to do something for the pipfile?

**Comment by hukkinj1** [comment]
I updated Pipfile and ran `pipenv update typing_extensions`. I'm not a pipenv user so please tell me if I did something wrong :)

File: Pipfile [hunk 1]
```diff
@@ -29,6 +29,6 @@ pathspec = ">=0.6"
 regex = ">=2020.1.8"
 toml = ">=0.10.1"
 typed-ast = "==1.4.2"
-typing_extensions = ">=3.7.4"
+typing_extensions = {"python_version <" = "3.8","version >=" = "3.7.4"}
```
Comment (cooperlees) [inline Pipfile:32]: Me either - So lets wait for someone else to test / review / confirm.

**Review by cooperlees — CHANGES_REQUESTED** [review]

File: Pipfile (hunk 1 above)
Comment (cooperlees) [reply Pipfile:32]: Thanks for this btw!

Comment (ichard26) [reply Pipfile:32]: Seemingly no?
```console
ichard26@acer-ubuntu:~/programming/testing$ git clone https://github.com/hukkinj1/black.git
Cloning into 'black'...
remote: Enumerating objects: 4561, done.
remote: Total 4561 (delta 0), reused 0 (delta 0), pack-reused 4561
Receiving objects: 100% (4561/4561), 4.62 MiB | 175.00 KiB/s, done.
Resolving deltas: 100% (3005/3005), done.
ichard26@acer-ubuntu:~/programming/testing$ cd black
ichard26@acer-ubuntu:~/programming/testing/black$ git switch conditional-typing-extensions 
Branch 'conditional-typing-extensions' set up to track remote branch 'conditional-typing-extensions' from 'origin'.
Switched to a new branch 'conditional-typing-extensions'
ichard26@acer-ubuntu:~/programming/testing/black$ pipenv install --dev --python=/opt/python3.9.1/bin/python3.9
Creating a virtualenv for this project…
Pipfile: /home/ichard26/programming/testing/black/Pipfile
Using /opt/python3.9.1/bin/python3.9 (3.9.1) to create virtualenv…
⠧ Creating virtual environment...created virtual environment CPython3.9.1.final.0-64 in 1199ms
  creator CPython3Posix(dest=/home/ichard26/.local/share/virtualenvs/black-iVRwJbcC, clear=False, global=False)
  seeder FromAppData(download=False, pip=bundle, setuptools=bundle, wheel=bundle, via=copy, app_data_dir=/home/ichard26/.local/share/virtualenv)
    added seed packages: pip==20.2.4, setuptools==50.3.2, wheel==0.35.1
  activators BashActivator,CShellActivator,FishActivator,PowerShellActivator,PythonActivator,XonshActivator

✔ Successfully created virtual environment! 
Virtualenv location: /home/ichard26/.local/share/virtualenvs/black-iVRwJbcC
Installing dependencies from Pipfile.lock (71f415)…
  🐍   ▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉ 74/74 — 00:03:41
To activate this project's virtualenv, run pipenv shell.
Alternatively, run a command inside the virtualenv with pipenv run.
ichard26@acer-ubuntu:~/programming/testing/black$ pipenv shell
Launching subshell in virtual environment…
ichard26@acer-ubuntu:~/programming/testing/black$  . /home/ichard26/.local/share/virtualenvs/black-iVRwJbcC/bin/activate
(black) ichard26@acer-ubuntu:~/programming/testing/black$ pip show typing-extensions
Name: typing-extensions
Version: 3.7.4.3
Summary: Backported and Experimental Type Hints for Python 3.5+
Home-page: https://github.com/python/typing/blob/master/typing_extensions/README.rst
Author: Guido van Rossum, Jukka Lehtosalo, Lukasz Langa, Michael Lee
Author-email: levkivskyi@gmail.com
License: PSF
Location: /home/ichard26/.local/share/virtualenvs/black-iVRwJbcC/lib/python3.9/site-packages
Requires: 
Required-by: mypy, aiohttp
```

The `pip show typing-extensions` should fail AFAIK.

Comment (hukkinj1) [reply Pipfile:32]: > `Required-by: mypy, aiohttp`

The Pipfile contains mypy and aiohttp which both have `typing_extensions` as an unconditional dependency. So I think that's expected of `pip show typing-extensions`.

Comment (ichard26) [reply Pipfile:32]: Cool and good catch. Let's just chalk it up to "collaborator shouldn't try to review at midnight" 😅 

My bad.

**Review by ichard26 — APPROVED** [review]

**Pull request merged**